datafusion/optimizer/src/optimizer.rs

```diff
@@ -294,7 +294,7 @@ impl Optimizer {
                     self.optimize_recursively(rule, &new_plan, config)
                         .and_then(|plan| {
                             if let Some(plan) = &plan {
-                                assert_schema_is_the_same(rule.name(), &new_plan, plan)?;
+                                assert_schema_is_the_same(rule.name(), plan, &new_plan)?;
```

Comment:
👍  I double checked the order

https://github.com/apache/arrow-datafusion/blob/b3fe6aa68adb644d275d8914b3802c153b4a3a27/datafusion/optimizer/src/optimizer.rs#L427-L431